### ISSUE TITLE
[ui5-middleware-cfdestination] header x-forwarded-proto contains two protocols for version >= 3.1.0

### ISSUE THREAD

**Issue by qualiture** [ui5-community/ui5-ecosystem-showcase#874]
When running in Business Application Studio, we use the ui5-middleware-cfdestination plugin for local testing of our UI5 app connecting to a CAP service.
The CAP service runs in one terminal and is in BAS exposed at localhost:4004, and the UI5 app runs in another terminal.

Prior to version 3.1.0, everything worked fine. However, since the move to http-proxy-middleware it does not work anymore. A request to the CAP service results in the following error:

```
node:internal/errors:490
    ErrorCaptureStackTrace(err);
    ^

TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:399:5)
    at URL.onParseError (node:internal/url:565:9)
    at new URL (node:internal/url:645:5)
    at new URL (node:internal/url:642:22)
    at /home/user/projects/hana_spc_ui/app/monitorquality/node_modules/ui5-middleware-cfdestination/lib/cfdestination.js:300:24
    at Gunzip.<anonymous> (/home/user/projects/hana_spc_ui/app/monitorquality/node_modules/http-proxy-middleware/dist/handlers/response-interceptor.js:24:57)
    at Gunzip.emit (node:events:513:28)
    at endReadableNT (node:internal/streams/readable:1359:12)
```

I did some investigation, and apparently the `x-forwarded-proto` header contains the following value: `https,http`. The URL that is trying to be constructed will then be `https,http://port8080-workspaces-ws-vphdm.eu20.applicationstudio.cloud.sap` and that is indeed an invalid URL.

I think it is a bug that this header contains two values

**Comment by petermuessig** [comment]
Release is in progress and should be there in 2mins

Update ui5-middleware-cfdestination: 3.3.7 => 3.3.8 and it should work now!

THX for reporting

**Comment by qualiture** [comment]
@petermuessig many thanks for the quick fix, much appreciated! 🙏

I now get an AxiosError now (`connect ECONNREFUSED ::1:80`) but I don't think that's related to this issue and most likely a misconfiguration of the destination on my end...

**Comment by petermuessig** [comment]
@qualiture - we could also sanitize this header in the proxy - let me know if I should do this...

**Comment by qualiture** [comment]
Hi @petermuessig, I don't think it has something to do with the header... if I compare the request headers (in the browser console) there is no difference between the 3.0.1 and 3.3.8 version of the cfdestination. I'm not sure about the headers sent from the proxy to the endpoint, I have to dive into that a bit more...

Strangely, GET and HEAD requests seem to work just fine (the initial GET of the $metadata returns with statuscode 200) but the subsequent POST $batch request fail with statuscode 500.

One strange thing I noticed when doing a GET request is that the uri of returned entities also contains the `https,http` combo, not sure if that is wrecking havoc here:

```
{
    "d": {
        "results": [
            {
                "Id": 51870,
                // etc
                "MoreData": {
                    "__deferred": {
                        "uri": "https,http://port8080-workspaces-ws-vphdm.eu20.applicationstudio.cloud.sap/<destination>/v2/<service>/MyEntity(Id='51870')/MoreData"
                    }
                },
                "__metadata": {
                    "type": "MYService.MyEntity",
                    "uri": "https,http://port8080-workspaces-ws-vphdm.eu20.applicationstudio.cloud.sap/<destination>/v2/<service>/MyEntity(Id='51870')"
                }
            },
// etc
```

**Comment by petermuessig** [comment]
One more try - the proto may come from the backend but we may sanitize this in the proxy middleware...

**Comment by petermuessig** [comment]
Give this version a try: `ui5-middleware-cfdestination: 3.3.8 => 3.3.9`

**Comment by qualiture** [comment]
@petermuessig You're a star! 🙏 Fix 3.3.9 works like a charm!

Many thanks for finding a solution, much appreciated!
